### ISSUE TITLE
Enable/Disable listeners

### ISSUE THREAD

**Issue by NickNish09** [Impostor/Impostor#176]
**TL;DR How do you guys enable/disable listeners in a plugin?**

Hello everyone,
I was trying to enable/disable a listener based on chat commands. I have this Event Listener class:

```
...
namespace Impostor.Plugins.Eragon.Mods
{
    public class ShapeShift : IEventListener
    {
        ...
        [EventListener]
        public void OnGameEnded(IGameEndedEvent e)
        {
            _logger.LogInformation($"Game has ended with ShapeShift ON.");
        }

        [EventListener]
        public void OnGameStarted(IGameStartedEvent e)
        {
            _logger.LogInformation($"Game is starting with ShapeShift ON.");
         }
    }
}
```

And I have a GameEventListener (like the example one in the docs), where I want to check if a user types something like:
**/gamemode shapeshift**, to enable the ShapeShift event listener. In the GameEventListener I tried to do this:

```
namespace Impostor.Plugins.Eragon.Handlers
{
    public class GameEventListener : IEventListener
    {
        ...
        [EventListener]
        public void OnPlayerChat(IPlayerChatEvent e)
        {
            string[] command = e.Message.Split(' ');
            if(command[0] == "/gamemode"){
                if(1 < command.Length){
                    if(command[1] == "shapeshift"){
                        _logger.LogInformation("Enabled shapeshift.");
                        _unregisterShapeShift = _eventManager.RegisterListener(new ShapeShift(_logger)));
                    } else {
                        _logger.LogInformation("Shapeshift being disabled.");
                        _unregisterShapeShift.Dispose();
                    }
            }
        }
    }
}
``` 
I was able to enable the plugin going to a custom game and typing **/gamemode shapeshift**, but when I tried to disable it (with the Dispose method) the ShapeShift event listener was still enabled.
Do this kind of approach works? Trying to Dispose() a registered listener to disable it? If not, how would you guys do such feature? Of enabling/disabling the listeners through chat commands.

Anyway, thanks for the attention, great work!

**Comment by AeonLucid** [comment]
Closed why?

**Comment by NickNish09** [comment]
I thought that the idea of the plugins where to listen everything while the server is running, and was up to me (the programmer) to decide which events will be triggered if certain flags are true.
So I did some flags to allow the EventListeners to be triggered, instead of trying to disable the listeners. I don't know if that is the normal behavior we should expect, but that is why I closed.
